app/src/main/java/tech/jpco/qen/model/SQL.kt

```diff
@@ -0,0 +1,76 @@
+package tech.jpco.qen.model
+
+import android.annotation.SuppressLint
+import android.content.Context
+import com.squareup.sqldelight.EnumColumnAdapter
+import com.squareup.sqldelight.android.AndroidSqliteDriver
+import com.squareup.sqldelight.runtime.rx.asObservable
+import com.squareup.sqldelight.runtime.rx.mapToOneNonNull
+import io.reactivex.Observable
+import io.reactivex.Single
+import tech.jpco.qen.Database
+import tech.jpco.qen.viewModel.DrawPoint
+import tech.jpco.qen.viewModel.TouchEventType
+import tech.jpco.qen.viewModel.iLogger
+
+class SQL private constructor(ctx: Context) : PagesRepository {
+    companion object : SingletonHolder<SQL, Context>(::SQL)
```

Comment:
I'm not sure this really needs to be a singleton, as an instance could just be created locally by the `MainActivity` and stored there for the duration of that object's lifecyle, but this holder is clever.

Reply:
I habitually create SQL databases as singletons, just in case. Also, it'd have to be tied to the lifecycle of the ViewModel, as it is the direct interactor and has a lifecycle greater than the Activity's.